ocaml/src/asd_server.ml

```diff
@@ -1249,7 +1251,13 @@ let execute_update : type req res.
                   (function
                     | Lwt.Canceled -> Lwt.fail Lwt.Canceled
                     | ConcurrentModification -> Lwt.return `Retry
-                    | exn -> Lwt.fail exn) >>= function
+                    | NegDiskUsage ->
+                       Lwt_log.fatal "negative disk usage detected." >>= fun () ->
+                       Lwt_extra2.flush_logging () >>= fun () ->
+                       let () = exit 666 in
+                       Lwt.return `Retry
```

Comment:
unification goes outside in, so the `Lwt.return ...` (how to escape that backtick in inline markdown?) is needed. `exit` is a regular function with an irregular result. In short: no warning.